osu.Game/Screens/Play/PlayerLoader.cs

```diff
@@ -0,0 +1,211 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Primitives;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Graphics.Transformations;
+using osu.Framework.Screens;
+using osu.Game.Beatmaps;
+using osu.Game.Database;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Screens.Backgrounds;
+using osu.Game.Screens.Menu;
+using OpenTK;
+
+namespace osu.Game.Screens.Play
+{
+    public class PlayerLoader : OsuScreen
+    {
+        private readonly Player player;
+        private OsuLogo logo;
+        private BeatmapMetadataDisplay info;
+
+        protected override BackgroundScreen CreateBackground() => new BackgroundScreenBeatmap(Beatmap);
+
+        public PlayerLoader(Player player)
+        {
+            ValidForResume = false;
+            this.player = player;
+
+            Children = new Drawable[]
+            {
+                logo = new OsuLogo
+                {
+                    Scale = new Vector2(0.15f),
+                    Interactive = false,
+                },
+            };
+
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            Add(info = new BeatmapMetadataDisplay(Beatmap)
+            {
+                Alpha = 0,
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+            });
+
+            player.Preload(Game);
+        }
+
+        protected override void OnEntering(Screen last)
+        {
+            base.OnEntering(last);
+
+            Background.FadeTo(0.4f, 250);
+
+            Content.ScaleTo(0.7f);
+            Content.ScaleTo(1, 750, EasingTypes.OutQuint);
+            Content.FadeInFromZero(500);
+
+            Delay(1000, true);
+
+            logo.MoveToOffset(new Vector2(0, -180), 500, EasingTypes.InOutExpo);
+            Delay(250, true);
+
+            info.FadeIn(500);
+
+            Delay(2000, true);
+
+            Content.ScaleTo(0.7f, 300, EasingTypes.InQuint);
+            Content.FadeOut(250);
+
+            Delay(250);
+
+            Schedule(() =>

```

Comment:
I think pretty much all `Schedule` calls need a comment on _why_ things need to be scheduled. Otherwise it's really hard to find out reading the code.